composer/trainer/trainer.py

```diff
@@ -1161,7 +1178,40 @@ def __init__(
         assert not isinstance(device_train_microbatch_size, str)
 
         # Distributed
-        if deepspeed_config is not None or fsdp_config is not None or dist.get_world_size() > 1:
+        if fsdp_config is not None:
```

Comment:
init function is getting too long. Put this in a helper function init_parallelism_config